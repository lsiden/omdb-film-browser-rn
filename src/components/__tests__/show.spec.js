import React from "react"
import renderer from "react-native-test-utils"
import { Text } from "react-native"

import Show from "../show"

const createWrapper = (props = {}) => renderer(<Show {...props} />)

const children = <Text>{"some content"}</Text>

it("does not render children when condition is falsy", () => {
  const wrapper = createWrapper({
    when: false,
    children,
  })
  expect(wrapper.toJSON()).toBeFalsy()
})

it("renders children when condition is truthy", () => {
  const wrapper = createWrapper({
    when: true,
    children,
  })
  expect(wrapper.toJSON()).toBeTruthy()
  expect(wrapper.toJSON()).toMatchSnapshot()
})
