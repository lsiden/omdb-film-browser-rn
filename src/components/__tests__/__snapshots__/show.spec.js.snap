// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders children when condition is truthy 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  some content
</Text>
`;
